flutter/lib/ui/geometry.dart

```diff
@@ -836,40 +1079,82 @@ class RRect {
     return w < h ? w : h;
   }
 
-  /// The point halfway between the left and right and the top and bottom edges of this rectangle.
+  /// The point halfway between the left and right and the top and bottom edges
+  /// of this rectangle.
   Point get center => new Point(left + width / 2.0, top + height / 2.0);
 
+  double _get_min(double min, double radius1, double radius2, limit) {
+    double sum = radius1 + radius2;
+    if (sum > limit && sum != 0.0) {
+      return math.min(min, limit / sum);
+    }
+    return min;
+  }
+
+  void _scaleRadii() {
+    if (_scaled == null) {
+      double scale = 1.0;
+      final List<double> scaled = new List.from(_value);
+
+      scale = _get_min(scale, scaled[11], scaled[5], height);
+      scale = _get_min(scale, scaled[4], scaled[6], width);
+      scale = _get_min(scale, scaled[7], scaled[9], height);
+      scale = _get_min(scale, scaled[8], scaled[10], width);
+
+      if (scale < 1.0) {
+        for (int i = 4; i < _kDataSize; i += 1)
+          scaled[i] *= scale;
+      }
+
+      _scaled = new RRect._fromList(scaled);
+    }
+  }
+
   /// Whether the given point lies inside the rounded rectangle.
```

Comment:
I think you're going to need to admit that the test depends on a "normalized" version of the rounded rect.